newIDE/app/src/BehaviorsEditor/EnumerateBehaviorsMetadata.js

```diff
@@ -30,16 +31,23 @@ export const enumerateBehaviorsMetadata = (
           behaviorType,
           behaviorMetadata: extension.getBehaviorMetadata(behaviorType),
         }))
-        .map(({ behaviorType, behaviorMetadata }) => ({
-          extension,
-          behaviorMetadata,
-          type: behaviorType,
-          defaultName: behaviorMetadata.getDefaultName(),
-          fullName: behaviorMetadata.getFullName(),
-          description: behaviorMetadata.getDescription(),
-          iconFilename: behaviorMetadata.getIconFilename(),
-          objectType: behaviorMetadata.getObjectType(),
-        }));
+        .map(({ behaviorType, behaviorMetadata }) =>
+          !behaviorMetadata.isPrivate() ||
+          (eventsFunctionsExtension &&
+            extension.getName() === eventsFunctionsExtension.getName())
```

Comment:
maybe you could extract this logic outside of the map, as a properly named variable, it will be easier to read.
Something like : 
`const isCurrentlyEditedEventsFunctionExtension = eventsFunctionsExtension && extension.getName() === eventsFunctionsExtension.getName()`

Reply:
You can use also a filter before the map, so you avoid to mix concerns in a map (slightly easier to read: "I see there is a filter, and then we map to transform the data").